File: 374-GuessNumber/guessNumber.java
/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int lowBound = 1;
        int highBound = n;
        int apiRslt = guess(lowBound + (highBound- lowBound)/2);
        while (apiRslt != 0) {
            if (apiRslt < 0) {
                highBound = lowBound + (highBound- lowBound)/2 -1;
            }
            if (apiRslt > 0) lowBound = lowBound + (highBound- lowBound)/2 +1;
            
            apiRslt = guess(lowBound + (highBound- lowBound)/2);
        }
        
        return lowBound + (highBound- lowBound)/2;
        
        
    }
}

File: 374-GuessNumber/guessNumber.cpp
// Forward declaration of guess API.
// @param num, your guess
// @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        int L=1;
        int R=n;
        int x=L+(R-L)/2;
        int res=guess(x);
        
        while(res!=0)
        {
            if(res==1)  L=x+1;
            else R=x-1;
            x=L+(R-L)/2;
            res=guess(x);
        }
        return x;
    }

};